src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift

```diff
@@ -512,7 +500,7 @@ final class RiskProvider: RiskProviding {
 	private func updateRiskProviderActivityState() {
 		if keyPackageDownloadStatus == .downloading || traceWarningDownloadStatus == .downloading {
 			self.updateActivityState(.downloading)
-		} else if (keyPackageDownloadStatus == .idle && (self.store.lastSuccessfulSubmitDiagnosisKeyTimestamp != nil || WarnOthersReminder(store: self.store).positiveTestResultWasShown)) &&
+		} else if keyPackageDownloadStatus == .idle && coronaTestService.hasAtLeastOneShownPositiveOrSubmittedTest &&
```

Comment:
Why is this check needed? Cant the RiskProvider be stuck in .downloading state with this check?

Reply:
This is here to get it out of the downloading state. If the risk calculation is running as normal, we don't want to switch back to idle here, because it changes to `.detecting` in `executeExposureDetection()`. But if we only download, and that is the case if `coronaTestService.hasAtLeastOneShownPositiveOrSubmittedTest` is true, then we switch back to idle here when both downloads are finished. This logic was actually introduced because without it the RiskProvider got stuck in the `.downloading` state when we were only downloading and not calculating the risk anymore.